azure-mgmt-graph-rbac/src/main/java/com/microsoft/azure/management/graphrbac/Application.java

```diff
@@ -0,0 +1,294 @@
+                WithMultiTenant {
+        }
+    }
+
+    /**
+     * Grouping of all the application update stages.
+     */
+    interface UpdateStages {
+        /**
+         * The stage of application update allowing specifying the sign on URL.
+         */
+        interface WithSignOnUrl {
+            /**
+             * Specifies the sign on URL.
+             *
+             * @param signOnUrl the URL where users can sign in and use this app
+             * @return the next stage in application update
+             */
+            Update withSignOnUrl(String signOnUrl);
+        }
+
+        /**
+         * The stage of application update allowing specifying reply URLs.
+         */
+        interface WithReplyUrl {
+            /**
+             * Adds a reply URL to the application.
+             *
+             * @param replyUrl URIs to which Azure AD will redirect in response to an OAuth 2.0 request
+             * @return the next stage in application update
+             */
+            Update withReplyUrl(String replyUrl);
+
+            /**
+             * Removes a reply URL.
+             *
+             * @param replyUrl the reply URL to remove
+             * @return the next stage in application update
+             */
+            Update withoutReplyUrl(String replyUrl);
+        }
+
+        /**
+         * The stage of application update allowing specifying identifier URLs.
+         */
+        interface WithIdentifierUrl {
+            /**
+             * Adds an identifier URL to the application.
+             *
+             * @param identifierUrl unique URI that Azure AD can use for this app
+             * @return the next stage in application update
+             */
+            Update withIdentifierUrl(String identifierUrl);
+
+            /**
+             * Removes an identifier URL from the application.
+             *
+             * @param identifierUrl identifier URI to remove
+             * @return the next stage in application update
+             */
+            Update withoutIdentifierUrl(String identifierUrl);
+        }
+
+        /**
+         * The stage of application update allowing specifying identifier keys.
+         */
+        interface WithCredential {
+            /**
+             * Starts the definition of a certificate credential.
+             * @param name the descriptive name of the certificate credential
+             * @return the first stage in certificate credential definition
+             */
+            CertificateCredential.UpdateDefinitionStages.Blank<Update> defineCertificateCredential(String name);
+
+            /**
+             * Starts the definition of a password credential.
+             * @param name the descriptive name of the password credential
+             * @return the first stage in password credential definition
+             */
+            PasswordCredential.DefinitionStages.Blank<Update> definePasswordCredential(String name);
```

Comment:
since this is in parent Update, shouldn't this be `PasswordCredential.*Update*DefinitionStages.Blank<Update>`?